File: src/code/js/_011.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    var max = 0, l = 0, r = height.length - 1;
    while (l < r) {
        max = Math.max(max, Math.min(height[l], height[r]) * (r - l));
        //舍弃短边 因为(l,r)之间任意一条边跟短边组成的面积要小于当前面积
        height[l] < height[r] ? l++ : r--;
    }
    return max;
};

File: src/code/java/_011.java
package code.java;

public class _011 {
    public int maxArea(int[] height) {
        int max = 0;
        int i = 0, j = height.length - 1;
        while (i < j) {
            int curr = Math.min(height[i], height[j]) * (j - i);
            max = Math.max(max, curr);
            // 舍弃短边 因为(i,j)之间任意一条边跟短边组成的面积要小于当前面积
            if (height[i] < height[j]) {
                i++;
            } else {
                j--;
            }
        }
        return max;
    }
}